### DISCUSSION TITLE
Reinforcement Learning

### DISCUSSION THREAD

**Discussion by luffy-Monkey123** [luffy-Monkey123/python-cool-programs#11] [Q&A] [answered]
What is reinforcement Learning ?

**Comment by RUTAGENGWA-ASANTE-BRUCE** [comment, marked as answer]
Reinforcement learning is a machine learning training method based on rewarding desired behaviors and/or punishing undesired ones. In general, a reinforcement learning agent is able to perceive and interpret its environment, take actions and learn through trial and error.

Hence, we can say that "Reinforcement learning is a type of machine learning method where an intelligent agent (computer program) interacts with the environment and learns to act within that." How a Robotic dog learns the movement of his arms is an example of Reinforcement learning.

What reinforcement learning is used for?
Reinforcement learning is a type of machine learning that enables a computer system to learn how to make choices by being rewarded for its successes. This can be an extremely powerful tool for optimization and decision-making. It's one of the most popular machine learning methods used today.
